### PR TITLE
#1089 Add CommonUtils.downloadUrl function

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1089)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-common/src/main/java/com/island/ohara/common/util/CommonUtils.java [hunk 1]
```diff
@@ -598,6 +598,24 @@ public static void copyURLToFile(
     }
   }
 
+  /**
+   * download file from url
+   *
+   * @param url where the file is download from
+   * @return the downloaded file
+   */
+  public static File downloadUrl(URL url) {
```
Comment (chia7712) [inline ohara-common/src/main/java/com/island/ohara/common/util/CommonUtils.java]: It would be better to have a variety accepting the timeout :)

Comment (saivirtue) [reply ohara-common/src/main/java/com/island/ohara/common/util/CommonUtils.java]: sure :)

File: ohara-common/src/main/java/com/island/ohara/common/util/CommonUtils.java [hunk 2]
```diff
@@ -598,6 +598,23 @@ public static void copyURLToFile(
...
 
+  /**
+   * download file from url
+   *
+   * @param url where the file is download from
+   * @param connectionTimeout the connection timeout
+   * @param readTimeout the read timeout
+   * @return the downloaded file
+   */
+  public static File downloadUrl(URL url, int connectionTimeout, int readTimeout) {
```
Comment (chia7712) [inline ohara-common/src/main/java/com/island/ohara/common/util/CommonUtils.java]: Could we use java.time.Duration to replace the **primitive type**??? Duration is more readable.

Comment (saivirtue) [reply ohara-common/src/main/java/com/island/ohara/common/util/CommonUtils.java]: Hmm...let me try it.

**Comment by saivirtue** [comment]
retry

File: ohara-streams/src/main/java/com/island/ohara/streams/StreamApp.java [hunk 3]
```diff
@@ -164,15 +166,7 @@ public static void main(String[] args) {
...
-    // create a tempFolder and new a file instance : /tmp/streamApp-XXXXX/streamApp.jar
-    File tmpFolder = CommonUtils.createTempFolder("streamApp-");
-    File outputFile = new File(tmpFolder, "streamApp.jar");
-
-    URL url = new URL(jarUrl);
-
-    // Download the jar
-    CommonUtils.copyURLToFile(url, outputFile, CONNECT_TIMEOUT, READ_TIMEOUT);
-    return outputFile;
+    return CommonUtils.downloadUrl(new URL(jarUrl), CONNECT_TIMEOUT, READ_TIMEOUT);
```
Comment (chia7712) [inline ohara-streams/src/main/java/com/island/ohara/streams/StreamApp.java]: Does compiler accept this code?

Comment (saivirtue) [reply ohara-streams/src/main/java/com/island/ohara/streams/StreamApp.java]: :(

**Pull request merged**